cumulusci/cli/cci.py

```diff
@@ -905,8 +906,17 @@ def org_scratch_delete(config, org_name):
 def task_list(config):
     data = []
     headers = ["task", "description"]
+    task_groups = OrderedDict()
     for task in config.project_config.list_tasks():
-        data.append((task["name"], task["description"]))
+        group = task["group"]
+        if group not in task_groups:
+            task_groups[group] = []
+        task_groups[group].append(task)
+    for group, tasks in task_groups.items():
+        data.append(("", ""))
+        data.append(("-- {} --".format(group), ""))
+        for task in tasks:
```

Comment:
`tasks = sorted(tasks, lambda x: x['name'])`

i think would really help, now that output isn't purely in YAML order.